jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -1,22 +1,112 @@
+/*
+ * Copyright (c) 2009-2021 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
 import com.jme3.anim.AnimationMask;
+import com.jme3.anim.Armature;
 import com.jme3.anim.tween.Tween;
 import com.jme3.util.clone.Cloner;
 import com.jme3.util.clone.JmeCloneable;
 
+/**
+ * Base implementation (1st level impl/prototype) of the interface #{@link Tween} for the new animation system.
+ * The Action class encloses an array of {@link Tween} and converts them into jme animation actions if they actually aren't.
+ *
+ * NB : this implementation mimics the {@link com.jme3.anim.tween.AbstractTween}, but it delegates the interpolation method #{@link Tween#interpolate(double)}
+ * to the #{@link BlendableAction} class.
+ *
+ * Subclasses need to make a constructor call passing in the Tween Actions to enclose them & override #{@link Action#interpolate(double)}.
+ *
+ * Example of operation(Uses Abstract Factory Pattern Method(interpolate/doInterpolate)) :
```

Comment:
To show a code that mimics the AbstractTween code, its an example for custom use, to do it in a one phase of abstraction layer, not 2 phases (as Nehon did).

But anyway normal users will not ever use this class in their code, they would rather use `ClipAction`, `BaseAction`, `BlendAction` and the `Tweens` utility class.

Reply:
If that's a debate about its usage, I will remove the example code.

Reply:
1. If the class were only for internal use, then it wouldn't need example code. Instead, it would need a comment like "internal use only". But I don't believe that's true.

2. Since the class isn't strictly for internal use, example code is a fine idea. But the example should illustrate a more typical use, such as testing the current state of a `Layer`:
```java
        // Determine the current action in the composer's walking layer.
        Action action = animComposer.getCurrentAction("walking layer");
        Action standAction = animComposer.action("stand");
        if (action == standAction) {
            // not traveling --- do something safe
        } else {
            // traveling --- do something exciting
        }
```

3. Or, if you prefer the current example, then please add some introductory explanation. A comment like "The following code sample implements an abstract Tween similar to AbstractTween" might help.

Reply:
I am not sure about that, but sounds a good idea anyway, I will rethink of a better testcase, but can I do this in a different PR ? So remove it for now and do it as an enhancement latter on ?

Reply:
@stephengold It's not that it's for internal use only but there are plenty of classes in JME that stand as example use-cases without including a (bad one) in the javadoc.  We could just link to them until we have a more succinct and less abstract/wrongy one.  (I get the impression that if I fully understood this example then I could point to a much better way to do what it's trying to do... that doesn't involve extending the Action class.  Hard to say.)

Reply:
@Scrappers-glitch and just to be clear, I'm not trying to put down the work you've been doing.  I think that it's great that you've taken on this task... but examples that are included in javadoc should be succinct and as perfect as possible.  (In fact, I'd argue that they could/should leave out the clone boiler plate, etc. and leave that to actual .java examples.)

Reply:
Well, clearly I get your point, no problem at all, I realized that better I do an example that would involve inheriting the whole pattern and not just a non-functioning class, may be we could have something later on the wiki for custom usages.

Reply:
Just in case, 
`
(I get the impression that if I fully understood this example then I could point to a much better way to do what it's trying to do... that doesn't involve extending the Action class. Hard to say.)`

The example I did, merges both the `BlendableAction` and `Action` classes, in a monophasic abstraction level..., but let's just remove this and focus on the real java docs for now.

Reply:
Removing the example seems like a good solution.